src/adapters/github.js

```diff
@@ -205,11 +206,81 @@ class GitHub extends Adapter {
   _getTree(path, opts, cb) {
     this._get(`/git/trees/${path}`, opts, (err, res) => {
       if (err) cb(err)
-      else cb(null, res.tree)
+      else {
+        const diff = this._getPatch(path)
+        const diffExists = (Object.keys(diff).length > 0)
+        const filteredTree = res.tree.filter((node) => {
+          node.patch = diff[node.path]
+          delete diff[node.path]
+          return !diffExists || node.patch
+        })
+
+        cb(null, filteredTree)
+      }
     })
   }
 
   // @override
+   _getPatch(path) {
+     const diff = {}
+     const files = $(".diff-view .file-info")
```

Comment:
That's a great suggestion. I'll take a stab at that change later tonight. Using the api also fixes the issue I'm seeing where filtering the tree based on the diff only works if you're actually on the "files" tab of a PR, but not on the rest of the PR. Thanks.